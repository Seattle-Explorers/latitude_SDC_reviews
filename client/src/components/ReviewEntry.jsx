import React from 'react';
import styled from 'styled-components';
import moment from 'moment';

const Wrapper = styled.div`
  margin: 25px 50px 25px 50px;  
`

const ShiftedWrapper = styled(Wrapper)`
  padding-left: 30px;
`

const TopStyle = styled.div`
  display: flex;
  margin-bottom: 10px;
`
const AvatarStyle = styled.img`
  width: 50px;
  height: 50px;
  border-radius: 50%;
  flex-direction: column;
  margin-right: 10px;
`
const TopText = styled.div`
  flex-direction: row
`
const Date = styled.span`
  color: #B3B3B3;
`


const ReviewEntry = (props) => {
  const { review, userDp, userName } = props;
  const reviewerURL = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${review.dp}.jpg`
  const date = moment(review.date).format('MMMM YYYY');

  const aReview = (
    <Wrapper>
      <TopStyle>
        <AvatarStyle src={reviewerURL} />
        <TopText>
          {review.reviewer_name}
          <br></br>
          <Date>{date}</Date>
        </TopText>
      </TopStyle>
      {review.body}
    </Wrapper>
  )

  if (review.respond){
    const listerURL = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${userDp}.jpg`;
    const aResponse = (
      <ShiftedWrapper>
        <TopStyle>
          <AvatarStyle src={listerURL} />
          <TopText>
            Response from {userName}
            <br></br>
            <Date>{date}</Date>
          </TopText>
        </TopStyle>
        {review.respond}
      </ShiftedWrapper>
    )

    return(
      <div>
        {aReview}
        {aResponse}
      </div>
    )
  }

  return (
    <div>
      {aReview}
    </div>
  )
}

export default ReviewEntry;