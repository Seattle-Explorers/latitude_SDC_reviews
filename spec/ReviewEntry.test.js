import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';

import ReviewEntry from '../client/src/components/ReviewEntry';

describe('Review Entry Component', () => {
  const reviewOne = {
    reviewer_name: 'Michael',
    dp: '11',
    date: '2016-07-19T19:33:37.736Z',
    body: 'here is some text',
  };

  const reviewTwo = {
    reviewer_name: 'Wonil',
    dp: '5',
    date: '2014-06-18T15:22:37.736Z',
    body: 'here is some text',
    respond: 'this is a response'
  };

  const userName = 'Josh';
  const userDp = '1'; 

  it('should render', () => {
    const component = <ReviewEntry review={reviewOne} userName={userName} userDp={userDp}/>;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should only render the reviewer avatar from s3 when there is no response', () => {
    const wrapperOne = mount(<ReviewEntry review={reviewOne} />);
    const urlReview = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${reviewOne.dp}.jpg`;

  });

  it('should render lister avatar and reviewer avatar when there is response', () => {
    const wrapperTwo = mount(<ReviewEntry review={reviewTwo} />);
    const urlReview = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${reviewTwo.dp}.jpg`;
    const urlUser = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${userDp}.jpg`

    console.log(wrapperTwo.find('AvatarStyle'));

    expect(wrapperTwo.find('AvatarStyle').first()).toHaveProp('src', urlReview);
    expect(wrapperTwo.find('AvatarStyle').second()).toHaveProp('src', urlUser);
  });

});