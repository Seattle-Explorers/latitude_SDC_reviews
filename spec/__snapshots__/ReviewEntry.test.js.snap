// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Entry Component should render 1`] = `
.c0 {
  margin: 25px 50px 25px 50px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 10px;
}

.c2 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 10px;
}

.c3 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  color: #B3B3B3;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <img
        className="c2"
        src="https://dteamdp.s3-us-west-2.amazonaws.com/pug11.jpg"
      />
      <div
        className="c3"
      >
        Michael
        <br />
        <span
          className="c4"
        >
          July 2016
        </span>
      </div>
    </div>
    here is some text
  </div>
</div>
`;
